pinot-server/src/main/java/com/linkedin/pinot/server/starter/helix/SegmentFetcherAndLoader.java

```diff
@@ -201,11 +201,10 @@ public void addOrReplaceOfflineSegment(String tableName, String segmentId, boole
             }
           }
         }
-        if (_segmentLoadMaxRetryCount <= retryCount) {
-          String msg =
-              "Failed to download segment " + segmentId + " (table " + tableName + " after " + retryCount + " retries";
-          LOGGER.error(msg);
-          throw new RuntimeException(msg);
+        if (retryCount == maxRetryCount) {
```

Comment:
I suppose this logic applies to whether we are throwing an exception during segment refresh vs. segment load. In the latter case, we want to retry multiple times (preferably retry only if download fails, not if load fails). 

So, can we try to name the variables accordingly? Is this a download or a refresh? Is this a network failure or a load failure? It seems that if segments fail to load, it does not make sense to retry.